network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -252,4 +264,20 @@ public String toString() {
         .toString();
     }
   }
+
+  private void saveRegisteredExecutors() throws IOException {
+    if (registeredExecutorFile != null) {
+      logger.info("Saving registered executors to {}", registeredExecutorFile);
+      ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(registeredExecutorFile));
```

Comment:
For paranoia, mostly, wouldn't it be better to write this to a temp file and the rename it? That avoids the hopefully rare case of the NM going down while the file is being written.